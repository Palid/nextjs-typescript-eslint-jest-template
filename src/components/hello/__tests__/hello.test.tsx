import React from "react";
import Hello from "../hello";
import TestRenderer from "react-test-renderer";
import { cleanup, render } from "@testing-library/react";

afterEach(cleanup);

describe("Index", () => {
  it("should match the snapshot", () => {
    const testRenderer = TestRenderer.create(<Hello name="Prottay Rudra" />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it("Should have the right content", () => {
    const { getByTestId } = render(<Hello name="Prottay Rudra" />);
    const title = getByTestId("title");
    expect(title).toHaveTextContent("Hello");
  });
});
