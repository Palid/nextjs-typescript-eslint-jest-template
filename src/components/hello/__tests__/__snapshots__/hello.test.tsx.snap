// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match the snapshot 1`] = `
<h1
  className="h1"
  data-testid="title"
>
  Hello
   
  Prottay Rudra
</h1>
`;
